### PR TITLE
Rebase onto E3SM master 101722

### PR DESCRIPTION
**Pull request by darincomeau**
Rebases onto E3SM master as of 10.17.22.

This PR points to a cime PR https://github.com/darincomeau/cime/pull/6 that has also been rebased on the current cime hash E3SM points to. After testing and that cime PR is merged into the cime/E3SMv2-Arctic branch, I'll do a new PR update the cime hash.

### CODE REVIEW

**Comment by darincomeau** [comment]
@milenaveneziani @apcraig this certainly needs testing to make sure machine support/PE layouts haven't been broken. It took 15 rounds of conflict resolution to do the rebase/merge, so hopefully I didn't mess something up in the process. I'll do a diff between the files I touched the most in this PR and what's currently on E3SMv2-Arctic as a check. I'll test on Cori tomorrow as well.

File: cime_config/allactive/config_pesall.xml [hunk 1]
```diff
@@ -2453,4 +2714,78 @@
       </pes>
     </mach>
   </grid>
+  <grid name="a%T62.+oi%oARRM60to10|a%TL319.+oi%oARRM60to10">
```
Comment (darincomeau) [inline cime_config/allactive/config_pesall.xml:2717]: @apcraig for example I'm not sure how these two layouts are new here, in particular the one for ARRM60to6 below. Maybe you had removed these at some point and I took the wrong change during conflict resolution?

**Comment by milenaveneziani** [comment]
@darincomeau: I will be testing this today on cori.

File: cime_config/allactive/config_pesall.xml (hunk 1 above)
Comment (apcraig) [reply cime_config/allactive/config_pesall.xml:2717]: Not sure either.  I haven't removed them.  I will create a sandbox and test as well.  May take me a day or two to get to it.

**Comment by darincomeau** [comment]
Great - thanks to both. If you want to use these land files for `WCYCL1950.arcticx4v1pg2_ARRM10to60E2r1` configuration, they're on the input server here:

 - add to `user_nl_elm` (replacing $DIN_LOC_ROOT with local inputdata directory):
` fsurdat = '$DIN_LOC_ROOT/lnd/clm2/surfdata_map/surfdata_arcticx4v1pg2_simyr1950_c221017.nc'`
` finidat = '$DIN_LOC_ROOT/lnd/clm2/initdata/20221017.ICRUELM-1950.arcticx4v1pg2_ARRM10to60E2r1.anvil.elm.r.0051-01-01-00000.nc'`

As I mentioned elsewhere these will be set as default for that configuration after this PR is resolved.

Edit: The E3SM PR for these land files is now opened here: https://github.com/E3SM-Project/E3SM/pull/5248

**Comment by apcraig** [comment]
So, I cannot create a case, getting

  ERROR: No machine onyx found

The other DoD machines seem to have the same problem.  I also noticed there are duplicate onyx, narwhal, blackbird, warhawk, and mustang entries in config_machines.xml and I'm not sure the allactive/config_pes.xml or components/mpas-ocean/cime_config/config_pes.xml is setup correctly for the DoD machines.  So something is going wrong in the process, not sure what.

What's the best way to proceed?  In some ways, I sort of prefer that this gets tested on cori then merged.  Then I'll create a PR to fix all of this.  I'll test on all the DoD machines again and make sure the various configurations work and we get reasonable pe layouts and things work consistent with the confluence "Running E3SM-Arctic" page.  

Alternatively, I could create a branch off @darincomeau's rebase-101722 branch, fix things, and then PR them back to his branch prior to the merge.  Either works OK for me.  Does anyone, @milenaveneziani, @darincomeau,  have a preference?  Even in this case, I think there is some benefit to having the cori testing done first, just so I know it works OK before I get too deep into testing on the DoD machines again.

**Comment by darincomeau** [comment]
Thanks for testing @apcraig , and sorry that DoD machine support is yet again broken. Hopefully it was something with the rebase that we can restore, rather than something introduced in the cime update. I did have to resolve conflicts in `config_machines.xml` *many* times. I also resolved conflicts in `cime_config/allactive/config_pesall.xml` many times, so it's certainly possible user error broke that, but I did not have to resolve conflicts in `components/mpas-ocean/cime_config/config_pes.xml`.

@milenaveneziani found a bug with introduced duplicate code in `components/mpas-ocean/src/shared/mpas_ocn_vmix_cvmix.F` (again, a file I did not touch during rebase), but she reports being able to submit a job after that fix. I'll push the fix shortly.

Rebasing has not gone smoothly these last couple of times and is getting more complicated - I'm wondering if we need to update this branch to sync with E3SM/master again, that maybe just a merge would be better?

**Comment by darincomeau** [comment]
@apcraig as far as how to proceed, I'll first push the fix I mentioned and then hopefully we can confirm this is working on Cori. I'd say let me look a little closer first before you create another branch to fix DoD support?

**Comment by darincomeau** [comment]
@apcraig I diff'ed the contents of the `onyx` entry in `cime_config/machines/config_machines.xml` in this PR and in the E3SMv2-Arctic branch, and they're the same, so it's not that. Also the `cmake_macros` files you've added are all still there.

The rebase removed the `cime_config/machines/config_compilers.xml` file. Could that be it? I don't know enough about cime's build system, but I believe that was being depreciated?

My only other thought is to revert the cime update in this PR back to the cime/E3SMv2-Arctic branch to see if that's responsible, but that's kind of a shot in the dark.

In short, after looking around a bit I'm not sure I have anything useful. On top of that I can't replicate @milenaveneziani 's successful build and submission on Cori...

**Comment by apcraig** [comment]
Thanks @darincomeau.  I did notice a while back double entries of a couple of the machines and I thought I took those out again, maybe I just did in my sandbox and never PR'ed it back.  I'm not sure what's going on.  Are we pushing the E3SMv2-Arctic mods onto master?  Then maybe when we rebase we get duplicates?  Again, not sure, but I wouldn't worry about how this happened at this point, lets just figure out how to move forward.  

It seems like the first thing is to make sure we can build and run on cori.  Then I can look at the DoD port as part of this PR or as a separate PR.  It's not a huge deal to review, fix, and test since we have recent working versions.  Hopefully CIME hasn't changed so much that we have to refactor a bunch of the port.  That's never fun, we'll see.  Finally, I'm glad config_compilers.xml is deleted, that shouldn't be causing any problems.

Let me know when you think it's a good time to start testing on the DoD machines.  Thanks.

**Comment by milenaveneziani** [comment]
I ran a successful build on cori and things are running now (in its 6 month of simulation).

For the sake of documentation, my hash is: `def6c0ecf3 (HEAD -> rebase/101722, darin/E3SM/rebase-101722) Merge remote-tracking branch 'origin/E3SM
v2-Arctic' into rebase-101722`
and this the run file I used (I know that some settings are redundant now):
```
#####################################################################
# Define user paths
#####################################################################

export USER=milena
export PROJECT=m1199
export MACHINE=cori-knl
export REPO_ROOT=/global/project/projectdirs/$PROJECT/$USER/E3SM-Model
export CASE_ROOT=/global/project/projectdirs/$PROJECT/$USER/E3SM-cases
export RUN_ROOT=/global/cscratch1/sd/$USER/e3sm_scratch/$MACHINE

#####################################################################
# Download repository
#####################################################################

# hash:
#def6c0ecf3 (HEAD -> rebase/101722, darin/E3SM/rebase-101722) Merge remote-tracking branch 'origin/E3SMv2-Arctic' into rebase-101722
export REPO_NAME=E3SM-Arctic-PR28
export E3SM_ROOT=$REPO_ROOT/$REPO_NAME

#####################################################################
# Define case:
#####################################################################

export COMPSET=WCYCL1950
export GRID=arcticx4v1pg2_ARRM10to60E2r1
export COMPILER=intel
export NOTE=baseline_bdvslat
export E3SM_CASE=`date +"%Y%m%d"`.${COMPSET}.${GRID}.${NOTE}.${MACHINE}

#####################################################################
# Create case
#####################################################################
cd $E3SM_ROOT/cime/scripts

./create_newcase \
-case $CASE_ROOT/$E3SM_CASE \
-compiler $COMPILER \
-mach $MACHINE \
-project $PROJECT \
-compset $COMPSET \
-res $GRID

cd $CASE_ROOT/$E3SM_CASE

#####################################################################
# PE_LAYOUT
#####################################################################

export NPROCS_CPL=6080
export NPROCS_ATM=6075
export NPROCS_LND=5952
export NPROCS_ROF=5952
export NPROCS_ICE=6000
export NPROCS_OCN=4096
export NPROCS_GLC=1
export NPROCS_WAV=1

export NTHRDS_CPL=1
export NTHRDS_ATM=1
export NTHRDS_LND=1
export NTHRDS_ROF=1
export NTHRDS_ICE=1
export NTHRDS_OCN=1
export NTHRDS_GLC=1
export NTHRDS_WAV=1

export ROOTPE_CPL=0
export ROOTPE_ATM=0
export ROOTPE_LND=0
export ROOTPE_ROF=0
export ROOTPE_ICE=0
export ROOTPE_OCN=6080
export ROOTPE_GLC=0
export ROOTPE_WAV=0

./xmlchange -file env_mach_pes.xml  -id NTASKS_ATM  -val $NPROCS_ATM
./xmlchange -file env_mach_pes.xml  -id NTASKS_LND  -val $NPROCS_LND
./xmlchange -file env_mach_pes.xml  -id NTASKS_ROF  -val $NPROCS_ROF
./xmlchange -file env_mach_pes.xml  -id NTASKS_ICE  -val $NPROCS_ICE
./xmlchange -file env_mach_pes.xml  -id NTASKS_OCN  -val $NPROCS_OCN
./xmlchange -file env_mach_pes.xml  -id NTASKS_GLC  -val $NPROCS_GLC
./xmlchange -file env_mach_pes.xml  -id NTASKS_WAV  -val $NPROCS_WAV
./xmlchange -file env_mach_pes.xml  -id NTASKS_CPL  -val $NPROCS_CPL

./xmlchange -file env_mach_pes.xml  -id NTHRDS_ATM  -val $NTHRDS_ATM
./xmlchange -file env_mach_pes.xml  -id NTHRDS_LND  -val $NTHRDS_LND
./xmlchange -file env_mach_pes.xml  -id NTHRDS_ROF  -val $NTHRDS_ROF
./xmlchange -file env_mach_pes.xml  -id NTHRDS_ICE  -val $NTHRDS_ICE
./xmlchange -file env_mach_pes.xml  -id NTHRDS_OCN  -val $NTHRDS_OCN
./xmlchange -file env_mach_pes.xml  -id NTHRDS_GLC  -val $NTHRDS_GLC
./xmlchange -file env_mach_pes.xml  -id NTHRDS_WAV  -val $NTHRDS_WAV
./xmlchange -file env_mach_pes.xml  -id NTHRDS_CPL  -val $NTHRDS_CPL

./xmlchange -file env_mach_pes.xml  -id ROOTPE_ATM  -val $ROOTPE_ATM
./xmlchange -file env_mach_pes.xml  -id ROOTPE_LND  -val $ROOTPE_LND
./xmlchange -file env_mach_pes.xml  -id ROOTPE_ROF  -val $ROOTPE_ROF
./xmlchange -file env_mach_pes.xml  -id ROOTPE_ICE  -val $ROOTPE_ICE
./xmlchange -file env_mach_pes.xml  -id ROOTPE_OCN  -val $ROOTPE_OCN
./xmlchange -file env_mach_pes.xml  -id ROOTPE_GLC  -val $ROOTPE_GLC
./xmlchange -file env_mach_pes.xml  -id ROOTPE_WAV  -val $ROOTPE_WAV
./xmlchange -file env_mach_pes.xml  -id ROOTPE_CPL  -val $ROOTPE_CPL

./xmlchange -file env_mach_pes.xml -id MAX_MPITASKS_PER_NODE -val 64

#####################################################################
# Setup and build
#####################################################################

./xmlchange -file env_build.xml -id CAM_TARGET -val theta-l
./xmlchange -file env_build.xml -id CAM_CONFIG_OPTS --append --val='-cosp'

./case.setup

#####################################################################
# Modify namelist files
#####################################################################

cat << EOF >> user_nl_eam
nhtfrq =   0,-24,-6,-6,-3,-24,0
mfilt  = 1,30,120,120,240,30,1
avgflag_pertape = 'A','A','I','A','A','A','I'
fexcl1 = 'CFAD_SR532_CAL', 'LINOZ_DO3', 'LINOZ_DO3_PSC', 'LINOZ_O3CLIM', 'LINOZ_O3COL', 'LINOZ_SSO3', 'hstobie_linoz'
fincl1 = 'extinct_sw_inp','extinct_lw_bnd7','extinct_lw_inp','CLD_CAL', 'TREFMNAV', 'TREFMXAV'
fincl2 = 'FLUT','PRECT','U200','V200','U850','V850','Z500','OMEGA500','UBOT','VBOT','TREFHT','TREFHTMN:M','TREFHTMX:X','QREFHT','TS','PS','TMQ','TUQ','TVQ','TOZ', 'FLDS', 'FLNS', 'FSDS', 'FSNS', 'SHFLX', 'LHFLX', 'TGCLDCWP', 'TGCLDIWP', 'TGCLDLWP', 'CLDTOT', 'T250', 'T200', 'T150', 'T100', 'T050', 'T025', 'T010', 'T005', 'T002', 'T001', 'TTOP', 'U250', 'U150', 'U100', 'U050', 'U025', 'U010', 'U005', 'U002', 'U001', 'UTOP', 'FSNT', 'FLNT'
fincl3 = 'PSL','T200','T500','U850','V850','UBOT','VBOT','TREFHT', 'Z700', 'TBOT:M'
fincl4 = 'FLUT','U200','U850','PRECT','OMEGA500'
fincl5 = 'PRECT','PRECC','TUQ','TVQ','QFLX','SHFLX','U90M','V90M'
fincl6 = 'CLDTOT_ISCCP','MEANCLDALB_ISCCP','MEANTAU_ISCCP','MEANPTOP_ISCCP','MEANTB_ISCCP','CLDTOT_CAL','CLDTOT_CAL_LIQ','CLDTOT_CAL_ICE','CLDTOT_CAL_UN','CLDHGH_CAL','CLDHGH_CAL_LIQ','CLDHGH_CAL_ICE','CLDHGH_CAL_UN','CLDMED_CAL','CLDMED_CAL_LIQ','CLDMED_CAL_ICE','CLDMED_CAL_UN','CLDLOW_CAL','CLDLOW_CAL_LIQ','CLDLOW_CAL_ICE','CLDLOW_CAL_UN'
fincl7 = 'O3', 'PS', 'TROP_P'
history_aerosol = .true.
EOF

cat << EOF >> user_nl_elm
hist_dov2xy = .true.,.true.
hist_fincl2 = 'H2OSNO', 'FSNO', 'QRUNOFF', 'QSNOMELT', 'FSNO_EFF', 'SNORDSL', 'SNOW', 'FSDS', 'FSR', 'FLDS', 'FIRE', 'FIRA'
hist_mfilt = 1,365
hist_nhtfrq = 0,-24
hist_avgflag_pertape = 'A','A'
fsurdat = '/global/cfs/cdirs/e3sm/inputdata/lnd/clm2/surfdata_map/surfdata_arcticx4v1pg2_simyr1950_c221017.nc'
finidat = '/global/cfs/cdirs/e3sm/inputdata/lnd/clm2/initdata/20221017.ICRUELM-1950.arcticx4v1pg2_ARRM10to60E2r1.anvil.elm.r.0051-01-01-00000.nc'
EOF

cat <<EOF >> user_nl_mpaso
config_GM_horizontal_taper = 'ramp'
config_GM_closure = 'Visbeck'
config_GM_spatially_variable_min_kappa = 400.0
config_GM_spatially_variable_max_kappa = 2000.0
config_Redi_horizontal_taper = 'ramp'
config_Redi_min_layers_diag_terms = 15
config_cvmix_background_scheme = 'LatDep'
config_cvmix_background_diffusion = 8.0e-5
config_cvmix_background_viscosity = 8.0e-4
config_use_mom_del2 = .true.
config_am_mocstreamfunction_enable = .true.
config_dt = '00:07:30'
EOF

cat <<EOF >> user_nl_mpassi
config_dynamics_subcycle_number = 2
config_ratio_ridging_work_to_pe = 21.3
config_dt = 900.0
EOF

./case.build

#####################################################################
# RUN_OPTIONS

./xmlchange -file env_run.xml -id ATM_NCPL -val 96
./xmlchange -file env_run.xml -id LND_NCPL -val 96
./xmlchange -file env_run.xml -id OCN_NCPL -val 96
./xmlchange -file env_run.xml -id ICE_NCPL -val 96

./xmlchange -file env_run.xml -id STOP_OPTION -val nyears
./xmlchange -file env_run.xml -id STOP_N -val 2
./xmlchange -file env_run.xml -id REST_OPTION -val nmonths
./xmlchange -file env_run.xml -id REST_N -val 6
./xmlchange -file env_workflow.xml -subgroup case.run -id JOB_QUEUE -val regular
./xmlchange -file env_workflow.xml -subgroup case.run -id JOB_WALLCLOCK_TIME -val "36:00:00"
```

**Comment by milenaveneziani** [comment]
@darincomeau: I think we can close this PR, right?

**Comment by darincomeau** [comment]
Closing as out of date.
